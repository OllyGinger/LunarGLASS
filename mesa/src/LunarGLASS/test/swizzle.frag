uniform float blend;
uniform vec4 u;

varying vec2 t;

void main()
{  
    float blendscale = 1.789;

    vec4 w = u;

    w.wy = t;
    
    gl_FragColor = mix(w, u, blend * blendscale);
}
