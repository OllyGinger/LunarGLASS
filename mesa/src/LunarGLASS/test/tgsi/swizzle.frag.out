; ModuleID = 'Top'

@gl_MaxLights = internal global i32 8
@gl_MaxClipPlanes = internal global i32 8
@gl_MaxTextureUnits = internal global i32 2
@gl_MaxTextureCoords = internal global i32 4
@gl_MaxVertexAttribs = internal global i32 16
@gl_MaxVertexUniformComponents = internal global i32 512
@gl_MaxVaryingFloats = internal global i32 32
@gl_MaxVertexTextureImageUnits = internal global i32 0
@gl_MaxCombinedTextureImageUnits = internal global i32 2
@gl_MaxTextureImageUnits = internal global i32 2
@gl_MaxFragmentUniformComponents = internal global i32 64
@gl_MaxDrawBuffers = internal global i32 2
@u = external addrspace(1) global <4 x float>
@p = external addrspace(1) global i1
@gl_FragColor = internal global <4 x float> zeroinitializer

define void @main() {
entry:
  %mix_retval = alloca <4 x float>
  %w = alloca <4 x float>
  %w_undef = alloca <4 x float>
  %w_dep = alloca <4 x float>
  %w_reorder = alloca <4 x float>
  %w2 = alloca <4 x float>
  %w_flow = alloca <4 x float>
  %w1 = load <4 x float> addrspace(1)* @u
  store <4 x float> %w1, <4 x float>* %w
  %w_dep2 = load <4 x float> addrspace(1)* @u
  store <4 x float> %w_dep2, <4 x float>* %w_dep
  %w_reorder3 = load <4 x float> addrspace(1)* @u
  store <4 x float> %w_reorder3, <4 x float>* %w_reorder
  %w_flow4 = load <4 x float> addrspace(1)* @u
  store <4 x float> %w_flow4, <4 x float>* %w_flow
  %0 = load <4 x float>* %w_reorder
  %w_reorder5 = insertelement <4 x float> %0, float 0x3FFC9FBE80000000, i32 2
  store <4 x float> %w_reorder5, <4 x float>* %w_reorder
  %t = call <2 x float> @llvm.gla.fReadInterpolant.v2f32(i32 0, i32 0)
  %1 = extractelement <2 x float> %t, i32 1
  %2 = insertelement <2 x float> undef, float %1, i32 0
  %3 = extractelement <2 x float> %t, i32 0
  %4 = insertelement <2 x float> %2, float %3, i32 1
  %5 = load <4 x float>* %w
  %6 = extractelement <2 x float> %4, i32 0
  %7 = insertelement <4 x float> %5, float %6, i32 1
  %8 = extractelement <2 x float> %4, i32 1
  %w6 = insertelement <4 x float> %7, float %8, i32 3
  store <4 x float> %w6, <4 x float>* %w
  %9 = load <4 x float>* %w_reorder
  %w_reorder7 = insertelement <4 x float> %9, float 0x3FFC9FBE80000000, i32 0
  store <4 x float> %w_reorder7, <4 x float>* %w_reorder
  %10 = load <4 x float> addrspace(1)* @u
  %11 = extractelement <4 x float> %10, i32 2
  %12 = insertelement <4 x float> undef, float %11, i32 0
  %13 = extractelement <4 x float> %10, i32 3
  %14 = insertelement <4 x float> %12, float %13, i32 1
  %15 = extractelement <4 x float> %10, i32 0
  %16 = insertelement <4 x float> %14, float %15, i32 2
  %17 = extractelement <4 x float> %10, i32 1
  %w28 = insertelement <4 x float> %16, float %17, i32 3
  store <4 x float> %w28, <4 x float>* %w2
  %18 = load <4 x float>* %w_reorder
  %w_reorder9 = insertelement <4 x float> %18, float 0x3FFC9FBE80000000, i32 1
  store <4 x float> %w_reorder9, <4 x float>* %w_reorder
  %19 = load <4 x float>* %w2
  %20 = extractelement <4 x float> %19, i32 0
  %21 = insertelement <2 x float> undef, float %20, i32 0
  %22 = extractelement <4 x float> %19, i32 2
  %23 = insertelement <2 x float> %21, float %22, i32 1
  %24 = load <4 x float>* %w_dep
  %25 = extractelement <2 x float> %23, i32 0
  %26 = insertelement <4 x float> %24, float %25, i32 0
  %27 = extractelement <2 x float> %23, i32 1
  %w_dep10 = insertelement <4 x float> %26, float %27, i32 1
  store <4 x float> %w_dep10, <4 x float>* %w_dep
  %t11 = call <2 x float> @llvm.gla.fReadInterpolant.v2f32(i32 0, i32 0)
  %28 = load <4 x float>* %w_dep
  %29 = extractelement <2 x float> %t11, i32 0
  %30 = insertelement <4 x float> %28, float %29, i32 2
  %31 = extractelement <2 x float> %t11, i32 1
  %w_dep12 = insertelement <4 x float> %30, float %31, i32 3
  store <4 x float> %w_dep12, <4 x float>* %w_dep
  %32 = load <4 x float> addrspace(1)* @u
  %33 = extractelement <4 x float> %32, i32 2
  %34 = insertelement <2 x float> undef, float %33, i32 0
  %35 = extractelement <4 x float> %32, i32 3
  %36 = insertelement <2 x float> %34, float %35, i32 1
  %37 = load <4 x float>* %w_undef
  %38 = extractelement <2 x float> %36, i32 0
  %39 = insertelement <4 x float> %37, float %38, i32 0
  %40 = extractelement <2 x float> %36, i32 1
  %w_undef13 = insertelement <4 x float> %39, float %40, i32 1
  store <4 x float> %w_undef13, <4 x float>* %w_undef
  %41 = load i1 addrspace(1)* @p
  br i1 %41, label %then, label %else

then:                                             ; preds = %entry
  %t14 = call <2 x float> @llvm.gla.fReadInterpolant.v2f32(i32 0, i32 0)
  %42 = extractelement <2 x float> %t14, i32 0
  %43 = load <4 x float>* %w_flow
  %w_flow15 = insertelement <4 x float> %43, float %42, i32 0
  store <4 x float> %w_flow15, <4 x float>* %w_flow
  br label %ifmerge

else:                                             ; preds = %entry
  %t16 = call <2 x float> @llvm.gla.fReadInterpolant.v2f32(i32 0, i32 0)
  %44 = extractelement <2 x float> %t16, i32 1
  %45 = load <4 x float>* %w_flow
  %w_flow17 = insertelement <4 x float> %45, float %44, i32 0
  store <4 x float> %w_flow17, <4 x float>* %w_flow
  br label %ifmerge

ifmerge:                                          ; preds = %else, %then
  %46 = load <4 x float>* %w_reorder
  %47 = load <4 x float>* %w_undef
  %48 = load <4 x float>* %w
  %49 = load <4 x float>* %w2
  %50 = fmul <4 x float> %48, %49
  %51 = load <4 x float>* %w_dep
  %52 = fmul <4 x float> %50, %51
  %53 = load <4 x float>* %w_flow
  %54 = fmul <4 x float> %52, %53
  %mix_retval18 = call <4 x float> @llvm.gla.fMix.v4f32.v4f32.v4f32.v4f32(<4 x float> %46, <4 x float> %47, <4 x float> %54)
  store <4 x float> %mix_retval18, <4 x float>* %mix_retval
  %gl_FragColor = load <4 x float>* %mix_retval
  store <4 x float> %gl_FragColor, <4 x float>* @gl_FragColor
  %55 = load <4 x float>* @gl_FragColor
  call void @llvm.gla.fWriteData.v4f32(i32 0, <4 x float> %55)
  ret void
}

declare <2 x float> @llvm.gla.fReadInterpolant.v2f32(i32, i32) nounwind readonly

declare <4 x float> @llvm.gla.fMix.v4f32.v4f32.v4f32.v4f32(<4 x float>, <4 x float>, <4 x float>) nounwind readnone

declare void @llvm.gla.fWriteData.v4f32(i32, <4 x float>) nounwind
; ModuleID = 'Top'

@u = external addrspace(1) global <4 x float>
@p = external addrspace(1) global i1

define void @main() {
entry:
  %w1 = load <4 x float> addrspace(1)* @u
  %t = call <2 x float> @llvm.gla.fReadInterpolant.v2f32(i32 0, i32 0)
  %0 = call <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.f32.v2f32.f32.v2f32(<4 x float> %w1, i32 10, float 0.000000e+00, i32 -1, <2 x float> %t, i32 1, float 0.000000e+00, i32 -1, <2 x float> %t, i32 0)
  %1 = call <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.v4f32.v4f32.v4f32.v4f32(<4 x float> undef, i32 15, <4 x float> %w1, i32 2, <4 x float> %w1, i32 3, <4 x float> %w1, i32 0, <4 x float> %w1, i32 1)
  %2 = call <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.f32.f32.f32.f32(<4 x float> %w1, i32 7, float 0x3FFC9FBE80000000, i32 -1, float 0x3FFC9FBE80000000, i32 -1, float 0x3FFC9FBE80000000, i32 -1, float 0.000000e+00, i32 -1)
  %3 = call <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.v4f32.v4f32.v2f32.v2f32(<4 x float> %w1, i32 15, <4 x float> %w1, i32 2, <4 x float> %w1, i32 0, <2 x float> %t, i32 0, <2 x float> %t, i32 1)
  %4 = call <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.v4f32.v4f32.f32.f32(<4 x float> undef, i32 3, <4 x float> %w1, i32 2, <4 x float> %w1, i32 3, float 0.000000e+00, i32 -1, float 0.000000e+00, i32 -1)
  %5 = load i1 addrspace(1)* @p
  br i1 %5, label %then, label %else

then:                                             ; preds = %entry
  %6 = call <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.v2f32.f32.f32.f32(<4 x float> %w1, i32 1, <2 x float> %t, i32 0, float 0.000000e+00, i32 -1, float 0.000000e+00, i32 -1, float 0.000000e+00, i32 -1)
  br label %ifmerge

else:                                             ; preds = %entry
  %7 = call <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.v2f32.f32.f32.f32(<4 x float> %w1, i32 1, <2 x float> %t, i32 1, float 0.000000e+00, i32 -1, float 0.000000e+00, i32 -1, float 0.000000e+00, i32 -1)
  br label %ifmerge

ifmerge:                                          ; preds = %else, %then
  %w_flow.0 = phi <4 x float> [ %6, %then ], [ %7, %else ]
  %8 = fmul <4 x float> %0, %1
  %9 = fmul <4 x float> %8, %3
  %10 = fmul <4 x float> %9, %w_flow.0
  %mix_retval18 = call <4 x float> @llvm.gla.fMix.v4f32.v4f32.v4f32.v4f32(<4 x float> %2, <4 x float> %4, <4 x float> %10)
  call void @llvm.gla.fWriteData.v4f32(i32 0, <4 x float> %mix_retval18)
  ret void
}

declare <2 x float> @llvm.gla.fReadInterpolant.v2f32(i32, i32) nounwind readonly

declare <4 x float> @llvm.gla.fMix.v4f32.v4f32.v4f32.v4f32(<4 x float>, <4 x float>, <4 x float>) nounwind readnone

declare void @llvm.gla.fWriteData.v4f32(i32, <4 x float>) nounwind

declare <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.v4f32.v4f32.f32.f32(<4 x float>, i32, <4 x float>, i32, <4 x float>, i32, float, i32, float, i32) nounwind readnone

declare <2 x float> @llvm.gla.fMultiInsert.v2f32.v2f32.v4f32.v4f32.f32.f32(<2 x float>, i32, <4 x float>, i32, <4 x float>, i32, float, i32, float, i32) nounwind readnone

declare <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.v4f32.v4f32.v2f32.v2f32(<4 x float>, i32, <4 x float>, i32, <4 x float>, i32, <2 x float>, i32, <2 x float>, i32) nounwind readnone

declare <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.f32.f32.f32.f32(<4 x float>, i32, float, i32, float, i32, float, i32, float, i32) nounwind readnone

declare <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.v4f32.v4f32.v4f32.v4f32(<4 x float>, i32, <4 x float>, i32, <4 x float>, i32, <4 x float>, i32, <4 x float>, i32) nounwind readnone

declare <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.f32.v2f32.f32.v2f32(<4 x float>, i32, float, i32, <2 x float>, i32, float, i32, <2 x float>, i32) nounwind readnone

declare <2 x float> @llvm.gla.fMultiInsert.v2f32.v2f32.v2f32.v2f32.f32.f32(<2 x float>, i32, <2 x float>, i32, <2 x float>, i32, float, i32, float, i32) nounwind readnone

declare <4 x float> @llvm.gla.fMultiInsert.v4f32.v4f32.v2f32.f32.f32.f32(<4 x float>, i32, <2 x float>, i32, float, i32, float, i32, float, i32) nounwind readnone

***Unsupported functionality: intrinsic in Bottom IR
compiling swizzle.frag...
uniform float blend ; 
uniform vec4 u ; 
uniform bool p ; 
varying vec2 t ; 
void  main (){
float blendscale = 1.789000 ; vec4 w = u ; vec4 w_undef ; vec4 w_dep = u ; vec4 w_reorder = u ; vec4 w2 = u ; vec4 w_flow = u ; w_reorder . z = blendscale ; w . wy = t ; w_reorder . x = blendscale ; w2 . xyzw = u . zwxy ; w_reorder . y = blendscale ; w_dep . xy = w2 . xz ; w_dep . zw = t ; w_undef . xy = u . zw ; if ( p ) w_flow . x = t . x ; else w_flow . x = t . y ; gl_FragColor = mix ( w_reorder , w_undef , w * w2 * w_dep * w_flow ) ; }



(
(declare (in ) vec2 t)
(declare (uniform ) bool p)
(declare (uniform ) vec4 u)
(declare (uniform ) float blend)
(declare (in ) vec4 gl_FragCoord)
(declare (in ) bool gl_FrontFacing)
(declare (out ) vec4 gl_FragColor)
(declare (out ) float gl_FragDepth)
(declare (in ) vec4 gl_Color)
(declare (in ) vec4 gl_SecondaryColor)
(declare (in ) float gl_FogFragCoord)
(declare (uniform ) mat4 gl_ModelViewMatrix)
(declare (uniform ) mat4 gl_ProjectionMatrix)
(declare (uniform ) mat4 gl_ModelViewProjectionMatrix)
(declare (uniform ) mat3 gl_NormalMatrix)
(declare (uniform ) mat4 gl_ModelViewMatrixInverse)
(declare (uniform ) mat4 gl_ProjectionMatrixInverse)
(declare (uniform ) mat4 gl_ModelViewProjectionMatrixInverse)
(declare (uniform ) mat4 gl_ModelViewMatrixTranspose)
(declare (uniform ) mat4 gl_ProjectionMatrixTranspose)
(declare (uniform ) mat4 gl_ModelViewProjectionMatrixTranspose)
(declare (uniform ) mat4 gl_ModelViewMatrixInverseTranspose)
(declare (uniform ) mat4 gl_ProjectionMatrixInverseTranspose)
(declare (uniform ) mat4 gl_ModelViewProjectionMatrixInverseTranspose)
(declare (uniform ) float gl_NormalScale)
(declare (uniform ) gl_LightModelParameters gl_LightModel)
(declare () int gl_MaxLights)
(declare () int gl_MaxClipPlanes)
(declare () int gl_MaxTextureUnits)
(declare () int gl_MaxTextureCoords)
(declare () int gl_MaxVertexAttribs)
(declare () int gl_MaxVertexUniformComponents)
(declare () int gl_MaxVaryingFloats)
(declare () int gl_MaxVertexTextureImageUnits)
(declare () int gl_MaxCombinedTextureImageUnits)
(declare () int gl_MaxTextureImageUnits)
(declare () int gl_MaxFragmentUniformComponents)
(declare (uniform ) (array mat4 4) gl_TextureMatrix)
(declare (uniform ) (array mat4 4) gl_TextureMatrixInverse)
(declare (uniform ) (array mat4 4) gl_TextureMatrixTranspose)
(declare (uniform ) (array mat4 4) gl_TextureMatrixInverseTranspose)
(declare (uniform ) gl_DepthRangeParameters gl_DepthRange)
(declare (uniform ) (array vec4 8) gl_ClipPlane)
(declare (uniform ) gl_PointParameters gl_Point)
(declare (uniform ) gl_MaterialParameters gl_FrontMaterial)
(declare (uniform ) gl_MaterialParameters gl_BackMaterial)
(declare (uniform ) (array gl_LightSourceParameters 8) gl_LightSource)
(declare (uniform ) gl_LightModelProducts gl_FrontLightModelProduct)
(declare (uniform ) gl_LightModelProducts gl_BackLightModelProduct)
(declare (uniform ) (array gl_LightProducts 8) gl_FrontLightProduct)
(declare (uniform ) (array gl_LightProducts 8) gl_BackLightProduct)
(declare (uniform ) (array vec4 2) gl_TextureEnvColor)
(declare (uniform ) (array vec4 4) gl_EyePlaneS)
(declare (uniform ) (array vec4 4) gl_EyePlaneT)
(declare (uniform ) (array vec4 4) gl_EyePlaneR)
(declare (uniform ) (array vec4 4) gl_EyePlaneQ)
(declare (uniform ) (array vec4 4) gl_ObjectPlaneS)
(declare (uniform ) (array vec4 4) gl_ObjectPlaneT)
(declare (uniform ) (array vec4 4) gl_ObjectPlaneR)
(declare (uniform ) (array vec4 4) gl_ObjectPlaneQ)
(declare (uniform ) gl_FogParameters gl_Fog)
(declare (in ) (array vec4 0) gl_TexCoord)
(declare () int gl_MaxDrawBuffers)
(declare (out ) (array vec4 2) gl_FragData)
(function main
  (signature void
    (parameters
    )
    (
      (declare () vec4 w_flow)
      (declare () vec4 w2)
      (declare () vec4 w_reorder)
      (declare () vec4 w_dep)
      (declare () vec4 w_undef)
      (declare () vec4 w)
      (assign (constant bool (1)) (xyzw) (var_ref w)  (var_ref u) ) 
      (assign (constant bool (1)) (xyzw) (var_ref w_dep)  (var_ref u) ) 
      (assign (constant bool (1)) (xyzw) (var_ref w_reorder)  (var_ref u) ) 
      (assign (constant bool (1)) (xyzw) (var_ref w_flow)  (var_ref u) ) 
      (assign (constant bool (1)) (z) (var_ref w_reorder)  (constant float (1.789000)) ) 
      (assign (constant bool (1)) (yw) (var_ref w)  (swiz yx (var_ref t) )) 
      (assign (constant bool (1)) (x) (var_ref w_reorder)  (constant float (1.789000)) ) 
      (assign (constant bool (1)) (xyzw) (var_ref w2)  (swiz zwxy (var_ref u) )) 
      (assign (constant bool (1)) (y) (var_ref w_reorder)  (constant float (1.789000)) ) 
      (assign (constant bool (1)) (xy) (var_ref w_dep)  (swiz xz (var_ref w2) )) 
      (assign (constant bool (1)) (zw) (var_ref w_dep)  (var_ref t) ) 
      (assign (constant bool (1)) (xy) (var_ref w_undef)  (swiz zw (var_ref u) )) 
      (if (var_ref p) (
        (assign (constant bool (1)) (x) (var_ref w_flow)  (swiz x (var_ref t) )) 
      )
      (
        (assign (constant bool (1)) (x) (var_ref w_flow)  (swiz y (var_ref t) )) 
      ))

      (declare (temporary ) vec4 mix_retval)
      (assign (constant bool (1)) (xyzw) (var_ref mix_retval)  (call mix ((var_ref w_reorder) (var_ref w_undef) (expression vec4 * (expression vec4 * (expression vec4 * (var_ref w) (var_ref w2) ) (var_ref w_dep) ) (var_ref w_flow) ) ))
) 
      (assign (constant bool (1)) (xyzw) (var_ref gl_FragColor)  (var_ref mix_retval) ) 
    ))

)


)